jenkins/bokchoy_db_pull_request.py

```diff
@@ -0,0 +1,290 @@
+    """
+    try:
+        repository.update_file(file_path, commit_message, contents, file_sha, branch_name)
+    except:
+        raise StandardError(
+            "Error updating database file: {}".format(file_path)
+        )
+
+
+def _create_file(repository, file_path, commit_message, contents, branch_name):
+    """
+    Create a commit on a branch that creates a new file with the string contents.
+    """
+    try:
+        repository.create_file(file_path, commit_message, contents, branch_name)
+    except:
+        raise StandardError(
+            "Error creating database file: {}".format(file_path)
+        )
+
+
+def _create_pull_request(repository, title, body, base, head):
+    """
+    Create a new pull request with the changes in head.
+    """
+    try:
+        pull_request = repository.create_pull(
+            title=title,
+            body=body,
+            base=base,
+            head=head
+        )
+    except:
+        raise StandardError(
+            "Could not create pull request"
+        )
+
+
+def _delete_branch(branch_object):
+    """
+    Delete a branch from a repository.
+    """
+    try:
+        branch_object.delete()
+    except:
+        raise StandardError(
+            "Failed to delete branch"
+        )
+
+
+@click.command()
+@click.option(
+    '--sha',
+    help="Sha of the merge commit to base the new PR off of",
+    required=True,
+)
+@click.option(
+    '--repo_root',
+    help="Path to local edx-platform repository that will "
+         "hold updated database files",
+    required=True,
+)
+def main(sha, repo_root):
+    logger.info("Authenticating with Github")
+    repository = _connect_to_repo("edx-platform")
+
+    fingerprint = _read_fingerprint_from_master(repository)
+    branch_name = "refs/heads/testeng/bokchoy_auto_cache_update"
+    if fingerprint:
+        branch_name = branch_name + "_" + fingerprint
+
+    if _branch_exists(repository, branch_name):
+        # If this branch already exists, then there's already a PR
+        # for this fingerprint. To avoid excessive PR's, exit.
+        logger.info(
+            "Branch name: {} already exists. Exiting."
+        )
+        return
+    branch_object = _create_branch(repository, branch_name, sha)
+    modified_files = _get_modified_files_list(repo_root)
+
+    changes_made = False
+    for db_file in BOKCHOY_DB_FILES:
+        repo_file_path = os.path.join('/', DB_CACHE_FILEPATH, db_file)
+        file_sha = _get_file_sha(repository, repo_file_path)
+        if file_sha:
+            if _file_has_changed(db_file, modified_files):
+                logger.info("File {} has changed.".format(repo_file_path))
+                local_file_data = _read_local_file_contents(repo_root, db_file)
+                logger.info("Updating database file: {}".format(repo_file_path))
+                _update_file(repository, repo_file_path, 'Updating migrations', local_file_data, file_sha, branch_name)
+                if not changes_made:
```

Comment:
nit: checking the prior value for changes_made is superfluous, in either case all we want is for its value to now be True.